src/zkevm_specs/evm/execution/mul.py

```diff
@@ -0,0 +1,66 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from ...util import FQ
+
+
+def mul(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    is_mul = (Opcode.DIV - opcode) * (Opcode.MOD - opcode) * FQ(8).inv()
+    is_div = (opcode - Opcode.MUL) * (Opcode.MOD - opcode) * FQ(4).inv()
+    is_mod = (opcode - Opcode.MUL) * (opcode - Opcode.DIV) * FQ(8).inv()
```

Comment:
8 and 4 would be constants in the circuit? If not, we need to check that they're indeed the value we presume.

Reply:
The opcode for MUL, DIV and MOD is 2, 4, 6. When the opcode is MUL, the value of `(Opcode.DIV - opcode) * (Opcode.MOD - opcode)` is 8. So we need to divide `(Opcode.DIV - opcode) * (Opcode.MOD - opcode)` by 8, which is equivalent to multiplying by inversion of 8. Probably I can explain it in the comment?

Reply:
added the comments to explain how we derive the constants.

Reply:
No, it was clear what they do. I was just wondering if we need to make them constants in the circuit or they should be witness values constrained to be equal to the values showed here.